Workiva/frugal
# Use titleServiceName for golang slim code generation

### Story:
There is a golang problem with how a struct or union field name is built into some references while using the slim flag when the field name is all caps and contains underscores. Generally, for such field names we hit the use of [this code](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L2258-L2266) which short circuits for all caps field names before doing [snakeToCamel](https://github.com/Workiva/frugal/blob/ec46850af057dde90b2a78c7d9a59888cf0d54a4/compiler/generator/golang/generator.go#L2226), which is the intended behavior. With the silm flag, we go past [this gate](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L926-L929) and directly to a [snakeToCamel call](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L937) (we do that [twice](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L950) actually) without first checking for all caps to match intended behavior. We can fix this by calling the existing helper function [titleServiceName](https://github.com/Workiva/frugal/blob/master/compiler/generator/golang/generator.go#L2253) which matches the current logic and adds the all caps check.

### Acceptance Criteria:

### Design Notes:
No design changes. This is a bug fix.

### How To Test:
- Confirm the test additions here expose the bug https://github.com/Workiva/frugal/pull/2106
- Confirm the PR of this branch into branch above shows problem is fixed https://github.com/Workiva/frugal/pull/2107
- Confirm newly added test in this work contains the fix shown in above two steps

### My Test Results:


#### Reviewers:
@Workiva/service-platform
